### ISSUE TITLE
Per-node DataTree chunking

### ISSUE THREAD

**Issue by sjperkins** [pydata/xarray#9634]
### Is your feature request related to a problem?

In the radio astronomy domain specific [xarray-ms](https://github.com/ratt-ru/xarray-ms), we construct a DataTree representing partitions of a legacy data format where each partition contains regular data cubes. As currently implemented, the custom backend supports a ``partition_chunks`` kwarg in the ``BackendEntrypoint.open_datatree`` method so that it is possible to specify different chunking schemas per partition:

- https://xarray-ms.readthedocs.io/en/latest/tutorial.html#per-partition-chunking

The chunking specification above is specific to a radio astronomy legacy format, but it may be more generally useful to be able to specify per-DataTree node  chunking.

### Describe the solution you'd like

Currently, ``BackendEntrypoint.open_datatree`` passes it's ``chunks`` kwarg to each ``Dataset`` constructor in the DataTree. This is quite coarse-grained as it applies the same chunking schema to all Datasets in the DataTree.

I propose that  the ``chunks`` kwarg in   ``BackendEntrypoint.open_datatree`` support a chunking dictionary per path (i.e. DataTree Node). For example:

```python
import xarray

xdt = xarray.open_datatree(..., chunks={
  "/path/to/node1": {"time": 20, "frequency": 16},
  "/path/to/a/node2": {"time": 10, "frequency": 4},
}
```

Then, when constructing Datasets in the DataTree, the chunking schema appropriate to the node can be applied.

An entry in the above dictionary does not necessarily need to only apply to a single node. It could also apply the chunking schema to each subtree below the node. But it may be better to make this more explicit

```python
xd = xarray.open_datatree(..., chunks={
  # Apply to node1 and any node below
  "/path/to/node1/...": {"time": 20, "frequency": 16}
}
```

### Describe alternatives you've considered

We've implemented a custom ``partition_chunks`` kwarg argument in the `BackendEntrypoint.open_datatree` method for our legacy data format.

**Comment by sjperkins** [comment]
Yes, in principle I'd like to submit a PR. Apologies for not replying, I need to devote more time to thinking about the change:

In particular, the ``open_datatree``  (and ``open_group_as_dict``) defers to the backend''s ``open_datatree``  implementation

https://github.com/pydata/xarray/blob/863184dfaa9c89b3f272da3042e5e9beaac200f3/xarray/backends/api.py#L859-L864

https://github.com/pydata/xarray/blob/863184dfaa9c89b3f272da3042e5e9beaac200f3/xarray/backends/api.py#L896-L901

which seems to imply that it's the backend's responsbility to interpret the `chunks` dictionary and pass it through to the backend's or xarray's method ``open_dataset`` . There doesn't immediately see a good way to do this by intercepting ``chunks`` before the API calls  and dispatching the appropriate chunking strategy/schema to each dataset.

Perhaps the full chunking schema/strategy could be passed to the ``open_dataset`` method, along with the tree node path? But that seems ugly.

Neither of the above seem appealing  -- I'll try find some more time to think about this.

**Comment by keewis** [comment]
I'm not sure if I didn't miss anything, but I don't think `open_datatree` does support `dask` / chunking at all, right now: nowhere in the code of the backends do we act on the chunks, and either way `open_dataset` calls [`_dataset_from_backend_dataset`](https://github.com/pydata/xarray/blob/863184dfaa9c89b3f272da3042e5e9beaac200f3/xarray/backends/api.py#L372-L414) after the call to `backend.open_dataset` to do that, so I think `open_datatree` should do something similar.

The resulting `_datatree_from_backend_datatree` would then also be the natural place for handling the per-group chunk arguments.
